proposals/lip-0051.md

```diff
@@ -814,484 +2596,147 @@ genesisTokenStoreSchema = {
-  ```
+```python
+storeKey = supplyEntry.tokenID
+storeValue = encode(
+    schema=supplyStoreSchema,
+    object={"totalSupply": supplyEntry.totalSupply}
+)
+```
 
 * For each entry `escrowEntry` in `genesisBlockAssetObject.escrowSubstore`, create an entry in the escrow substore with:
-  ```python
-  storeKey = uint32be(escrowEntry.escrowChainID) || uint32be(escrowEntry.localID)
-  storeValue = {"amount": escrowEntry.amount} serialized using escrowStoreSchema.
-  ```
+```python
+storeKey = escrowEntry.tokenID
+storeValue = encode(
+    schema=escrowStoreSchema,
+    object={"amount": escrowEntry.amount}
+)
+```
 
-* Create an entry in the available local ID substore with:
-  ```python
-  storeKey = EMPTY_BYTES
-  storeValue = {"nextAvailableLocalID": genesisBlockAssetObject.availableLocalIDSubstore.nextAvailableLocalID}
-               serialized using availableLocalIDStoreSchema.
-  ```
+* For each entry `supportedTokensEntry` in `genesisBlockAssetObject.supportedTokensSubstore`, create an entry in the supported tokens substore with:
+```python
+storeKey = supportedTokensEntry.chainID
+storeValue = encode(
+    schema=supportedTokensSchema,
+    object={"supportedTokenIDs": [tkID for each tkID in supportedTokensEntry.supportedTokenIDs]}
+)
+```
 
-* For each entry `terminatedChainID` in `genesisBlockAssetObject.terminatedEscrowSubstore`, create an entry in the terminated escrow substore with:
-  ```python
-  storeKey = uint32be(terminatedChainID)
-  storeValue = {"escrowTerminated": True} serialized using terminatedEscrowSchema.
-  ```
 
 Once the module store is initialized, its validity is attested via the two checks below.
 
 * Check that for each native token the total supply is correct.
   This can be done by checking that the function below returns `True`.
   ```python
-  validateSupplyStoreEntries():
+  def validateSupplyStoreEntries() -> bool:
       computedSupplies = {}
       for (storeKey, storeValue) in user substore:
-          chainID = storeKey[20:24] # Part of the key corresponding to the chain ID of the token
-          localID = storeKey[24:28] # Part of the key corresponding to the local ID of the token
-          if chainID == CHAIN_ID_ALIAS_NATIVE:
-              computedSupplies[localID] += sum of availableBalance and all locked amounts of storeValue
+          tokenID = storeKey[ADDRESS_LENGTH:ADDRESS_LENGTH + TOKEN_ID_LENGTH] # Part of the key corresponding to the token ID of the token
+          if getChainID(tokenID) == OWN_CHAIN_ID:
+              computedSupplies[tokenID] += sum of availableBalance and all locked amounts of storeValue
 
       for (storeKey, storeValue) in escrow substore:
-          localID = storeKey[4:8] # Part of the key corresponding to the local ID of the token
-          computedSupplies[localID] += storeValue.amount
+          tokenID = storeKey[:8] # Part of the key corresponding to the token ID of the token
```

Comment:
key for escrow is tokenID so no need to slice here?

Reply:
Good catch. Perhaps this was not updated at all since we don't even use the proper constant. In fact, the escrow substore has keys `escrowedChainID + tokenID`, so I think it should be 

`tokenID = storeKey[CHAIN_ID_LENGTH:]`